packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -2022,6 +2022,47 @@ void main() {
 
     semantics.dispose();
   });
+
+  testWidgets('NavigationRailDestination padding properly applied', (WidgetTester tester) async {
+    const EdgeInsets defaultPadding = EdgeInsets.symmetric(horizontal: 8.0);
+    const EdgeInsets secondItemPadding = EdgeInsets.symmetric(vertical: 30.0);
+    const EdgeInsets thirdItemPadding = EdgeInsets.symmetric(horizontal: 10.0);
+
+    await _pumpNavigationRail(
+      tester,
+      navigationRail: NavigationRail(
+        labelType: NavigationRailLabelType.all,
```

Comment:
Could you also add a test for NavigationRailLabelType.selected and NavigationRailLabelType.none? 

Might also be worth adding a test for `NavigationRailLabelType.all` and `NavigationRailLabelType.selected` making sure that the default padding is applied (`const EdgeInsets.symmetric(horizontal: _horizontalDestinationPadding`) when padding is not defined.

Reply:
Yeah sure I'll add a test for NavigationRailLabelType.selected and NavigationRailLabelType.none. Also, in the added test the first destination item uses default padding so it's checking for the default padding ```const EdgeInsets defaultPadding = EdgeInsets.symmetric(horizontal: 8.0);```.